images/microOS/config.xml

```diff
@@ -560,31 +118,20 @@
         <type
             image="iso"
             primary="true"
-            flags="dmsquash"
+            flags="overlay"
             hybridpersistent_filesystem="ext4"
             hybridpersistent="true"
-            kernelcmdline="quiet systemd.show_status=yes console=ttyS0,115200 console=tty0 net.ifnames=0 \$ignition_firstboot ignition.platform.id=qemu"
-            bootkernel="custom"
-            devicepersistency="by-uuid"
-        >
-            <systemdisk>
-                <volume name="home"/>
-                <volume name="root"/>
-                <volume name="tmp"/>
-                <volume name="opt"/>
-                <volume name="srv"/>
-                <volume name="boot/grub2/i386-pc"/>
-                <volume name="boot/grub2/x86_64-efi" mountpoint="boot/grub2/x86_64-efi"/>
-                <volume name="boot/writable"/>
-                <volume name="usr/local"/>
-                <volume name="var" copy_on_write="false"/>
-            </systemdisk>
-            <machine memory="512" guestOS="suse" HWversion="4">
-                <vmdisk id="0" controller="ide"/>
-                <vmnic driver="e1000" interface="0" mode="bridged"/>
-            </machine>
-        </type>
+            kernelcmdline="quiet systemd.show_status=yes console=ttyS0,115200 console=tty0"
+            devicepersistency="by-uuid"/>
     </preferences>
+    <users>
+        <user password="suse1234" home="/root" name="root" groups="root" pwdformat="plain"/>
+        <user password="suse1234" home="/home/vagrant" name="vagrant" pwdformat="plain"/>
```

Comment:
This feels something that we're never going to be using on a live/installation image. I understand you need it there to write the authorized_keys, but I don't think that's the way to go here. I don't think this is something we want to manage from a live boot image, which will force us to clean up this user on every single boot.

Reply:
oh now I see why you think differently ....from my experience I create default user for liveiso and installation as well so user can skip the so call "firstboot" process to input anything or adduser etc to reduce the adoption difficulty. My thought was if we are going to setup vagrant user may as well using the same. We can change it but I think we need to setup user by default for all images.

Reply:
Assuming we can change a user's password on first boot and *persist* that, I think we can have a default user to allow the debugging and other maintenance tasks should one want/need to. I think that is fine. In that case I'd add an `aquarium` user, with passwordless sudoers rights, and would not set up a root password at all.

As for `vagrant`, I really think this should only be available if we are running a Vagrant image - otherwise it makes little sense to have it around, even if just from a cosmetic point of view. But, more importantly, we can't be having a default `authorized_keys` with a ssh key that is known to everyone -- that is simply a security risk we can't afford on a real system. It's fine for a vagrant image running locally for testing, but not if we are running Aquarium on production.

Does that make sense?

Reply:
Yeah totally. I understand that. However, let listed all our images. 
1 and 2 Both Install image and LiveISO should able to persistent password and configuration. 
So user will update their password after install and using liveiso for long term and even production. (Discussion 1)  
3 vagrant using ssh key also needed to be around and require to update once user wanted to use for long term. (Discussion 2) 

For install and live iso both either require users to create the "first-boot" process to make a system working. However, that also makes it a little bit more difficult to use. To me they are two different approaches but not security-related which is how easy can someone with no background able to download and run it. 

Appliance, download, start iso or dd USB, boot, running. 
VS 
Start LiveISO configuration, ask questions, setting .... etc then Running. 

Meanwhile, this discussion can also apply to all the configurations being default to different settings e.g.  I use DHCP for the first nic, if we have two interfaces, then I create a bond with two different subnets automatically to build a front and back subnet etc. 

I can do either way, but pre-configuration default can be 1 of the multiple configurations we pick for Quick Start. And user can easily adopt and understand. Alternative may require us to create an installation process that may require lots of work we don't want to maintain in the long run. ( right now assume we can leverage yast tools which is ok, but that fail back to different distribution user may not used to the interface for system configuration tool)

Reply:
Discussion 2  I agreed I can create an aquarium user who is a better user than a vagrant user. 
However, I was just thinking in terms of the regular users my ask ...why do you have multiple users for different images type and not the same as 1 
So they need to remember which user to change the password after install and which platform using which user etc. 
Therefore, I don't mind adding aquarium user for LiveISO and Install Profile. But adding more image type may end up adding more user role, should we consolidate into a single user. 
For the root password I could disable it, I was testing the root password because microOS disable it and sudo didn't work. So I will remove the root login after user and sudo is working.

Reply:
> 1 and 2 Both Install image and LiveISO should able to persistent password and configuration.
> So user will update their password after install and using liveiso for long term and even production. (Discussion 1)

Cool - I think that way we can indeed have a very basic default password (e.g., `aquarium`) for a default user `aquarium`, and have us change that through the frontend before bootstrapping. In the mean time, the user that use `aquarium:aquarium` however they see fit.

> 3 vagrant using ssh key also needed to be around and require to update once user wanted to use for long term. (Discussion 2)

I don't think the user will ever use the same image from Vagrant to a live system. They are two different images, and one will be running under vagrant and the other will be used on an actual machine. I don't think there's, nor should there be, a path from Vagrant -> Live ISO. So I don't think this is something we should care about.


> For install and live iso both either require users to create the "first-boot" process to make a system working. However, that also makes it a little bit more difficult to use. To me they are two different approaches but not security-related which is how easy can someone with no background able to download and run it.
> 
> Appliance, download, start iso or dd USB, boot, running.
> VS
> Start LiveISO configuration, ask questions, setting .... etc then Running.

I think this is a fair point. We should make the experience as easy to the user as possible. Having a default user/password is not the end of the world, assuming we can indeed change it and persist it -- which I'm assuming is the case based on your previous comment.

> Meanwhile, this discussion can also apply to all the configurations being default to different settings e.g. I use DHCP for the first nic, if we have two interfaces, then I create a bond with two different subnets automatically to build a front and back subnet etc.

For now, let's assume that the user can configure those things manually through the Live image. We may want to allow them to configure that, at some point, through the installer, but that's a stretch goal that is still a bit far away :)


> I can do either way, but pre-configuration default can be 1 of the multiple configurations we pick for Quick Start. And user can easily adopt and understand. Alternative may require us to create an installation process that may require lots of work we don't want to maintain in the long run. ( right now assume we can leverage yast tools which is ok, but that fail back to different distribution user may not used to the interface for system configuration tool)

I'm not following here: do you mean have pre-built images with bonding enabled and such?